odoo/sql_db.py

```diff
@@ -8,77 +8,42 @@
 the ORM does, in fact.
 """
 
-from contextlib import contextmanager
-from datetime import timedelta, datetime
-import itertools
 import logging
 import os
-import time
+import re
 import threading
+import time
 import uuid
 import warnings
-
+from contextlib import contextmanager
+from datetime import datetime, timedelta
 from inspect import currentframe
+
 import psycopg2
-import psycopg2.extras
 import psycopg2.extensions
+import psycopg2.extras
 from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT, ISOLATION_LEVEL_READ_COMMITTED, ISOLATION_LEVEL_REPEATABLE_READ
 from psycopg2.pool import PoolError
 from psycopg2.sql import SQL, Identifier
 from werkzeug import urls
 
+from . import tools
 from .tools.func import frame_codeinfo, locked
 
 psycopg2.extensions.register_type(psycopg2.extensions.UNICODE)
 
-_logger = logging.getLogger(__name__)
-_logger_conn = _logger.getChild("connection")
-
-real_time = time.time.__call__  # ensure we have a non patched time for query times when using freezegun
-
-def unbuffer(symb, cr):
-    if symb is None:
+def undecimalize(value, cr):
+    if value is None:
         return None
-    return str(symb)
-
-def undecimalize(symb, cr):
-    if symb is None:
-        return None
-    return float(symb)
-
-psycopg2.extensions.register_type(psycopg2.extensions.new_type((700, 701, 1700,), 'float', undecimalize))
-
-
-from . import tools
-
-from .tools import parse_version as pv
-if pv(psycopg2.__version__) < pv('2.7'):
```

Comment:
@ryv-odoo the commit msg / PR desc seem to imply that this code was specific to *Python 2.7*? That's not right, the code is a compatibility shim for *psycopg2* versions before 2.7 (e.g. 2.6.2).

I don't really mind removing that shim for v16, as we've been requiring 2.7+ for years, but you should be aware of why you're actually doing it (and adapt the commit message).

Reply:
Yep, I just fail my commit message.